backend/data/processors/conditions_preprocessing.py

```diff
@@ -432,6 +442,134 @@ def map_word_to_program_type(processed, regex_word, type):
+    
+
+    # Final 2 terms note
+    processed, number_of_subs = re.subn(
+        r"It is strongly recommended that students only complete this course within their final 2 terms of study", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Recommended to take this course within their final 2 terms of study."
+
+    # Business school students notes
+    processed, number_of_subs = re.subn(
+        r"Only available to single and double degree Business School students in Term 1. It will be offered to non-Business School students in Terms 2 and 3.<br/><br/>", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Only available to single and double degree Business School students in Term 1. It will be offered to non-Business School students in Terms 2 and 3."
+    
+    processed, number_of_subs = re.subn(
+    r"Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Terms 1 and 3.<br/><br/>", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Terms 1 and 3."
+    
+    processed, number_of_subs = re.subn(
+    r"Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3.<br/><br/>", r"", processed, flags=re.IGNORECASE
+    ) 
+    if number_of_subs > 0:
+        note += "Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3.<br/><br/>"
+
+    # Application and progression check notes
+    processed, number_of_subs = re.subn(
+    r"This course is by application only. Please contact the Co-op office for more information", r"", processed, flags=re.IGNORECASE
+    ) 
+    if number_of_subs > 0:
+        note += "This course is by application only. Please contact the Co-op office for more information"
+
+    processed, number_of_subs = re.subn(
+    r"This course is by application only.", r"", processed, flags=re.IGNORECASE
+    ) 
+    if number_of_subs > 0:
+        note += "This course is by application only."
+
+    processed, number_of_subs = re.subn(
+    r"Its recommended to seek a progression check prior to application.", r"", processed, flags=re.IGNORECASE
+    ) 
+    if number_of_subs > 0:
+        note += "It is recommended to do a progression check prior to enrolling."
+
+    processed, number_of_subs = re.subn(
+    r"This course is by application only. Please visit Business School website for more information<br/><br/>", r"", processed, flags=re.IGNORECASE
+    ) 
+    if number_of_subs > 0:
+        note += "This course is by application only. Please visit Business School website for more information<br/><br/>"
```

Comment:
avoid keeping the HTML tags. Ideally, we should run this function after they are removed.

Reply:
actually yeah isn't this function being run after HTML deletion? HTML deletion is one of the first stages in the processing pipeline so this will probably mean that the `re.subn` calls won't match anything

Reply:
removed, will move it to after the remove HTML function is run and check everything is ok